### PR TITLE
Remove max_rows parameter from DataFrame.to_markdown function

### PR DESCRIPTION
**Pull request by beobest2**
Resolves https://github.com/databricks/koalas/issues/1498

I removed the `max_rows` parameter from the `DataFrame.to_markdown` function.

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -1764,14 +1764,10 @@ def to_latex(
...
         )
 
-    def to_markdown(self, buf=None, mode=None, max_rows=None):
+    def to_markdown(self, buf=None, mode=None):
         """
         Print DataFrame in Markdown-friendly format.
 
-        .. note:: This method should only be used if the resulting Pandas object is expected
-                  to be small, as all the data is loaded into the driver's memory. If the input
-                  is large, set max_rows parameter.
```
Comment (itholic) [inline databricks/koalas/frame.py:1773]: We should notice users that this API loads the all data into the driver's memory.

So, you can remove the only `max_rows` part:

```python
        .. note:: This method should only be used if the resulting Pandas object is expected
                  to be small, as all the data is loaded into the driver's memory.
```

Comment (beobest2) [reply databricks/koalas/frame.py]: okay. I will recover that part.

**Comment by itholic** [comment]
Seems good

**Comment by HyukjinKwon** [comment]
Looks good but let's hold off for now. Breaking changes should ideally happen in Koalas 1.0.0 and we're discussing about the timing offline.

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -1764,13 +1764,12 @@ def to_latex(
             kdf._to_internal_pandas(), self.to_latex, pd.DataFrame.to_latex, args
         )
 
-    def to_markdown(self, buf=None, mode=None, max_rows=None):
+    def to_markdown(self, buf=None, mode=None):
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py:1767]: Wait .. `to_markdown` is only supported with pandas 1.0.0. @itholic, was `max_rows` supported in pandas 1.0? Seems that parameter was never supported from the first place (https://github.com/pandas-dev/pandas/commit/7ab73a93c94e6557e5d77b2966549955e5c51cbc). Why was this added at https://github.com/databricks/koalas/commit/f44022927f3336ce666f82faa07d7276ed28de05?

Comment (itholic) [reply databricks/koalas/frame.py:1767]: @HyukjinKwon No, It was just my mistake and I didn't notice that.

Comment (HyukjinKwon) [reply databricks/koalas/frame.py:1767]: That's alright. Wanted to double check. Let's merge this as a bug fix.
